src/js/utils/date-utils.js

```diff
@@ -0,0 +1,283 @@
+/**
+ * Create a formatted, locale-aware string representation of a datetime.
+ *
+ * These functions are meant to act as a general localization function
+ * for server-provided UTC datetimes, creating UI-friendly strings in the user's
+ * local timezone formatted to the user's preferred language
+ *
+ * Most of the heavy lifting is done by the 'moments' js library,
+ * with the moment-with-locales and moment-timezones add-ons.
+ *
+ * By sending preferences data, even in an incomplete state, we can generate the appropriate
+ * datetime string for surfacing to the user.
+ *
+ * @module DateUtils
+ */
+
+
+/*
+DATE FORMATS:
+These were previously called from edx-platform/common/djangoapps/util/date_utils.py,
+but in an effort to move this to the front-end, they are redefined here and translated into
+local-aware moment.js formats.
+
+DEFAULT_SHORT_DATE_FORMAT
+example: Jan 01, 2016
+python: "%b %d, %Y"
+
+DEFAULT_LONG_DATE_FORMAT
+example: Friday, January 01, 2016
+python: "%A, %B %d, %Y"
+
+DEFAULT_TIME_FORMAT
+example: Jan 01, 2016 at 15:30
+python: "%b %d, %Y at %H:%M"
+
+DEFAULT_DATE_TIME_FORMAT
+example: Jan 01, 2016 15:30 UTC
+python: "%b %d, %Y %H:%M"
+*/
+
+var DATE_FORMATS = {
+    DEFAULT_SHORT_DATE_FORMAT: 'll',
+    DEFAULT_LONG_DATE_FORMAT: 'LLLL',
+    DEFAULT_TIME_FORMAT: 'LTS z',
+    DEFAULT_DATE_TIME_FORMAT: 'll HH[:]mm z'
+};
+
+
+(function(define) {
+    'use strict';
+    define([
+        'jquery',
+        'edx-ui-toolkit/js/utils/string-utils',
+        'moment',
+        'momenttimezone'
+    ], function(
+        $,
+        StringUtils,
+        moment,
+        momentTZ
+        ) {
+        var DateUtils, isValid, dateObject, userTimezone, localizeTime, timeDisplay, localize;
+
+        /**
+         * Helper function to determine variable/context validity.
+         *
+         * @param {string} candidateVariable to test
+         * @returns {boolean}
+         */
+        isValid = function(candidateVariable) {
+            var valid = true;
+            if (candidateVariable === undefined) {
+                valid = false;
+            } else if (candidateVariable === 'undefined') {
+                valid = false;
+            } else if (candidateVariable === '') {
+                valid = false;
+            } else if (candidateVariable === 'Invalid date') {
+                valid = false;
+            } else {
+                valid = true;
+            }
+            return valid;
```

Comment:
In general, you don't need to have conditionals that return booleans, because the conditional expressions themselves are the result you want. In addition, you can benefit from all the ways in which JavaScript strings are falsey (http://stackoverflow.com/questions/19839952/all-falsey-values-in-javascript).

I believe you can change this whole thing to:

    return candidateVariable && candidateVariable !== 'Invalid date';

I'd also make 'Invalid date' into a constant.